Added cron job for test sessions automating deletion

PlayJeri opened the pull request:
### Brief description

Added a cron job to delete expired testing sessions.

### Change list

- Added nestjs schedule package to handle the cron jobs.
- Added cron job and method to box service to delete the ended sessions.
- Added unit tests for the new method

PlayJeri commented:
@MikhailDeriabin 

I wasn't quire sure about the implementation but this is what the new code does.
Every day at midnight the deleteBoxById is called for every box that has the sessionStage set at "End"

MikhailDeriabin commented:
@PlayJeri 

The cron job should use the timestamps in the box data, not the stage:

1. All the boxes should be removed based on the `boxRemovalTime` field
2. All the session data should be reset based on the `sessionResetTime` field

PlayJeri commented:
### Brief description

This pull request introduces a new feature for scheduling and resetting testing sessions, along with some dependency updates. The most important changes include the addition of the `BoxScheduler` class, new test cases for the scheduler, and updates to the module imports and providers.

### Change list

#### New Feature: Box Scheduler

* [`src/box/box.scheduler.ts`](diffhunk://#diff-fd7d8562b79fb82a32b853d72cc3b9e0d020b1f0b644221880684296fb813085R1-R71): Added a new `BoxScheduler` class that uses cron jobs to reset testing sessions by deleting or resetting boxes based on their expiration times.
* [`src/box/box.service.ts`](diffhunk://#diff-8bf43102f5c273643f01c13636469c8020763bd83535865026637de5933798e8R363-R381): Added a new method `getExpiredBoxes` to retrieve boxes that have expired based on their removal or reset times.

#### Test Cases

* [`src/__tests__/box/BoxScheduler/resetTestingSessions.test.ts`](diffhunk://#diff-9376d9a9d1f2e07ea931e2823ab45af667499dac7ae52ab4643285804ed37c2fR1-R297): Added comprehensive test cases for the `BoxScheduler.resetTestingSessions` method to ensure it correctly deletes or resets boxes based on their expiration times.

#### Module Updates

* [`src/__tests__/box/modules/box.module.ts`](diffhunk://#diff-c4d7fc33fd2d868a7ce58555911de9fa04354c1d5c438b04417d189d6d23256aR14): Updated the `BoxModule` to include the `BoxScheduler` in its providers and added a new method to get the `BoxScheduler` instance. [[1]](diffhunk://#diff-c4d7fc33fd2d868a7ce58555911de9fa04354c1d5c438b04417d189d6d23256aR14) [[2]](diffhunk://#diff-c4d7fc33fd2d868a7ce58555911de9fa04354c1d5c438b04417d189d6d23256aR67-R71)
* [`src/__tests__/box/modules/boxCommon.ts`](diffhunk://#diff-07de98ece1b3d5aef1e4ee328f5c999f1ceaff68542121bbb52e9004468ee75fR28): Updated the `BoxCommonModule` to include the `BoxScheduler` in its providers. [[1]](diffhunk://#diff-07de98ece1b3d5aef1e4ee328f5c999f1ceaff68542121bbb52e9004468ee75fR28) [[2]](diffhunk://#diff-07de98ece1b3d5aef1e4ee328f5c999f1ceaff68542121bbb52e9004468ee75fL59-R60)
* [`src/app.module.ts`](diffhunk://#diff-089f4f2474b64391c42b6e66aed33977e132058d92108f0a63234a7862e1f8b8R33): Imported `ScheduleModule` from `@nestjs/schedule` and added it to the module imports to enable scheduling capabilities. [[1]](diffhunk://#diff-089f4f2474b64391c42b6e66aed33977e132058d92108f0a63234a7862e1f8b8R33) [[2]](diffhunk://#diff-089f4f2474b64391c42b6e66aed33977e132058d92108f0a63234a7862e1f8b8R52)
* [`src/box/box.module.ts`](diffhunk://#diff-709f02ffb1e9995cccec1e1e8c1c3399c8d98d6f15061a218e7404c4709794d7R29): Included `BoxScheduler` in the providers array to ensure it is available for dependency injection. [[1]](diffhunk://#diff-709f02ffb1e9995cccec1e1e8c1c3399c8d98d6f15061a218e7404c4709794d7R29) [[2]](diffhunk://#diff-709f02ffb1e9995cccec1e1e8c1c3399c8d98d6f15061a218e7404c4709794d7L46-R54)

#### Dependency Updates

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R34): Added `@nestjs/schedule` to the dependencies to support the new scheduling feature.

MikhailDeriabin reviewed (approved):
Looks nice
